tokenizers/src/tokenizer/encoding.rs

```diff
@@ -543,21 +603,51 @@ impl std::iter::FromIterator<(u32, String, (usize, usize), Option<u32>, u32)> fo
     }
 }
 
+#[inline]
+fn split_off_back<T>(vec: &mut Vec<T>, at: usize) -> Vec<T> {
+    assert!(vec.len() >= at);
+    let mut other = Vec::with_capacity(at);
+    let left_over_len = vec.len() - at;
+    let at_isize = at.try_into().unwrap();
+    unsafe {
```

Comment:
Ouch, not sure we need `unsafe` in this library :)

Reply:
I think staying away from unsafe is probably best, even if it means 1 clone. That's my opinion at least.